src/cow-react/api/proxy/hooks.ts

```diff
@@ -15,8 +17,19 @@ const SUPPORTED_CHAINS: Partial<Record<Chain, SupportedChainId>> = {
 
 const UNSUPPORTED_CHAIN_ID = null
 
-function hasSupportedChainId(token: any): token is FetchTokensResult {
-  return token?.chainId !== UNSUPPORTED_CHAIN_ID
+function isValidFetchTokensResult(token: any): token is FetchTokensResult {
+  if (typeof token !== 'object') {
+    return false
+  }
+
+  const hasValidChainId =
+    (token.chainId !== UNSUPPORTED_CHAIN_ID &&
+      typeof token.chainId === 'number' &&
+      token.chainId === SupportedChainId.MAINNET) ||
```

Comment:
If you are checking it this is mainnet, do you also need to check it's not unsupported?
Maybe the parenthesis is at the wrong spot?

Reply:
You are absolutely right - fixing it.